### PR TITLE
Fix breakage caused by changes in formulaic

### PR DESCRIPTION
**Pull request by stanmart**
It seems like formulaic might be considering the `EncodedTermStructure` a private class based on the fact that it is being moved in a non-major release. A bigger refactor of the `formula` module might be worth considering at some point now that formulaic reached 1.0 and is more stable. Until then, this PR provides a quick fix for #422.

Checklist
* [ ] Added a `CHANGELOG.rst` entry

### CODE REVIEW

File: src/tabmat/formula.py [hunk 1]
```diff
@@ -24,6 +23,11 @@
 from .sparse_matrix import SparseMatrix
 from .split_matrix import SplitMatrix
 
+try:
+    from formulaic.materializers.base import EncodedTermStructure
+except ImportError:
+    from formulaic.materializers.types.formula_materializer import EncodedTermStructure
```
Comment (lbittarello) [inline src/tabmat/formula.py:29]: Could we use the `formulaic` version to select the import path instead of a `try/except` block?

Comment (stanmart) [reply src/tabmat/formula.py:29]: Unfortunately I don't know in which version the change will appear. It's on main now, and my guess is `1.1`, but I'm not sure.

**Review by lbittarello — APPROVED** [review]
Maybe raise an issue with `formulaic` about breaking the public API without a major release?

**Comment by MarcAntoineSchmidtQC** [comment]
@stanmart, I hijacked your PR to also update a CI runner so that the CI would pass.

**Comment by MarcAntoineSchmidtQC** [comment]
oops, I thought it would be a simple fix. Not the case.

I will open a different PR to fix this.

**Review by MarcAntoineSchmidtQC — APPROVED** [review]

**Comment by stanmart** [comment]
haha, all good, I'll merge it then

**Pull request merged**